picard/ui/searchdialog.py

```diff
@@ -0,0 +1,362 @@
+                qdur=str(metadata.length / 2000),
+                isrc=metadata['isrc'],
+                limit=25)
+
+    def handle_reply(self, document, http, error):
+        if error:
+            error_msg = _("Unable to fetch results. Close the dialog and try "
+                    "again. See debug logs for more details.")
+            self.show_error(error_msg)
+            return
+
+        try:
+            tracks = document.metadata[0].recording_list[0].recording
+        except (AttributeError, IndexError):
+            error_msg = _("No results found. Please try a different search query.")
+            self.show_error(error_msg)
+            return
+
+        if self.file_:
+            tmp = sorted((self.file_.orig_metadata.compare_to_track(
+                track, File.comparison_weights) for track in tracks),
+                reverse=True,
+                key=itemgetter(0))
+            tracks = [item[3] for item in tmp]
+
+        del self.search_results[:]  # Clear existing data
+        self.parse_tracks_from_xml(tracks)
+        self.display_results()
+
+    def display_results(self):
+        self.show_table(self.table_headers)
+        for row, obj in enumerate(self.search_results):
+            track = obj[0]
+            table_item = QtGui.QTableWidgetItem
+            self.table.insertRow(row)
+            self.table.setItem(row, 0, table_item(track.title))
+            self.table.setItem(row, 1, table_item(track.length))
+            self.table.setItem(row, 2, table_item(track.artist))
+            self.table.setItem(row, 3, table_item(track.release))
+            self.table.setItem(row, 4, table_item(track.date))
+            self.table.setItem(row, 5, table_item(track.country))
+            self.table.setItem(row, 6, table_item(track.release_type))
+
+    def parse_tracks_from_xml(self, tracks_xml):
+        for node in tracks_xml:
+            rec_id = node.id
+            rec_title = node.title[0].text
+            artist = artist_credit_from_node(node.artist_credit[0])[0]
+            try:
+                length = format_time(node.length[0].text)
+            except AttributeError:
+                length = ""
+            try:
+                releases = node.release_list[0].release
+                for release in releases:
+                    rel_id = release.id
+                    rel_title = release.title[0].text
+                    if "date" in release.children:
+                        date = release.date[0].text
+                    else:
+                        date = None
+                    if "country" in release.children:
+                        country = release.country[0].text
+                    else:
+                        country = ""
+                    rg = release.release_group[0]
+                    rg_id = rg.id
+                    types_list = []
+                    if "primary_type" in rg.children:
+                        types_list.append(rg.primary_type[0].text)
```

Comment:
Those can be translated (same for secondary types below, but using `release_group_secondary_type` as the context).

```python
from picard.i18n import ugettext_attr
# ...
types_list.append(ugettext_attr(rg.primary_type[0].text, 'release_group_primary_type'))
```